public/class-award-force-sso.php

```diff
@@ -84,17 +93,35 @@ private function requestSlugByEmail($email)
      * @param $slug
      * @return mixed
      */
-    private function requestAuthToken($slug)
+    private function requestAuthToken($slug, WP_User $user)
     {
+        if ($token = $this->sendAuthTokenRequest($slug)->auth_token) {
+            return $token;
+        }
+
+        $slug = $this->getSlug($user, true);
         $retries = 5;
+
         while ($retries > 0) {
-            $response = $this->api->get('/user/' . $slug . '/auth-token');
-            if ($token = $response->auth_token) {
-                return $token;
+            if ($response = $this->sendAuthTokenRequest($slug)) {
+                if ($token = $response->auth_token) {
+                    return $token;
+                }
+                $this->api->handleException(new Exception($response->message));
```

Comment:
An exception will break out of the retry loop? Is that what you want here?

Reply:
Yes, retry only if we don't obtain a response. In case of error, log the message and abort